### PR TITLE
Add TAPEX blog post

### PR DESCRIPTION
**Pull request by SivilTaram**

### CODE REVIEW

File: tapex.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        <img class="avatar avatar-user" src="https://aeiljuispo.cloudimg.io/v7/https://s3.amazonaws.com/moonup/production/uploads/1652435645515-612ee6a7b960e78c6d2319d4.jpeg?w=200&h=200&f=face" title="Gravatar">
+        <div class="bfc">
+            <code>SivilTaram</code>
+            <span class="fullname">Qian Liu</span>
+        </div>
+    </a>
+</div>
+
+
+In recent years, language model pre-training has achieved a great success via leveraging large-scale textual data. By employing pre-training tasks such as [masked language modeling](https://arxiv.org/abs/1810.04805), these models have demonstrated surprising performance on several downstream tasks. However, the dramatic gap between the pre-training task (e.g., language moedling) and the downstream task (e.g., table question answering) makes existing pre-training not efficient enough. In practice, we often need an *extremely large amount* of pre-training corpus to obtain promising improvement, even for [domain-adaptive pretraining](https://arxiv.org/abs/2004.02349). How might we design a pre-training task to close the gap, and thus accelerate pre-training?
```
Comment (NielsRogge) [inline tapex.md:29]:
```suggestion
In recent years, language model pre-training has achieved great success via leveraging large-scale textual data. By employing pre-training tasks such as [masked language modeling](https://arxiv.org/abs/1810.04805), these models have demonstrated surprising performance on several downstream tasks. However, the dramatic gap between the pre-training task (e.g., language moedling) and the downstream task (e.g., table question answering) makes existing pre-training not efficient enough. In practice, we often need an *extremely large amount* of pre-training corpus to obtain promising improvement, even for [domain-adaptive pretraining](https://arxiv.org/abs/2004.02349). How might we design a pre-training task to close the gap, and thus accelerate pre-training?
```

File: tapex.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+![corpus](assets/74_tapex/tapex-performance.png)
+
+
+### Comparison to Previous Table Pre-training
+
+The earliest work on table pre-training, [TAPAS](https://aclanthology.org/2020.acl-main.398/) from Google Research and [TaBERT](https://aclanthology.org/2020.acl-main.745/) from Meta AI, have revealed that collecting more *domain-adaptive* data can improve the downstream performance. However, these previous work mainly employ *general-purpose* pre-training tasks, e.g., language modeling or its variants. TAPEX explores a different path by sacrificing the naturalness of the pre-trained source in order to obtain a *domain-adaptive* pre-trained task, i.e. SQL execution. A graphical comparison of BERT, TAPAS/TaBERT and our TAPEX can be seen below.
```
Comment (NielsRogge) [inline tapex.md:109]:
```suggestion
The earliest work on table pre-training, [TAPAS](https://aclanthology.org/2020.acl-main.398/) from Google Research - also [available in 🤗 Transformers](https://huggingface.co/docs/transformers/model_doc/tapas) - and [TaBERT](https://aclanthology.org/2020.acl-main.745/) from Meta AI, have revealed that collecting more *domain-adaptive* data can improve the downstream performance. However, these previous works mainly employ *general-purpose* pre-training tasks, e.g., language modeling or its variants. TAPEX explores a different path by sacrificing the naturalness of the pre-trained source in order to obtain a *domain-adaptive* pre-trained task, i.e. SQL execution. A graphical comparison of BERT, TAPAS/TaBERT and our TAPEX can be seen below.
```

File: tapex.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+
+![comparsion](assets/74_tapex/comparsion-tapex.png)
+
+We believe the SQL execution task is closer to the downstream table question answering task, especially from the perspective of structural reasoning capabilities. Imagine you are faced with a SQL query `SELECT City ORDER BY Year` and a natural question `Sort all cities by year`. The reasoning paths required by the SQL query and the question are similar, except that SQL is a bit more rigid than the natural language. If a language model can be pre-trained to faithfully “execute” SQL queries and produce correct results, it should have a deep understanding on natural language with similar intents.
```
Comment (NielsRogge) [inline tapex.md:114]:
```suggestion
We believe the SQL execution task is closer to the downstream table question answering task, especially from the perspective of structural reasoning capabilities. Imagine you are faced with a SQL query `SELECT City ORDER BY Year` and a natural question `Sort all cities by year`. The reasoning paths required by the SQL query and the question are similar, except that SQL is a bit more rigid than natural language. If a language model can be pre-trained to faithfully “execute” SQL queries and produce correct results, it should have a deep understanding on natural language with similar intents.
```

File: tapex.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+![efficiency](assets/74_tapex/tapex-efficiency.png)
+
+What about the efficiency? How efficient is such a pre-training method compared to the previous pre-training? The answer is given in the above figure: compared with previous table pre-training method TaBERT, TAPEX could yield 2% improvement only using 2% of the pre-training corpus, achieving a speedup of nearly **50** times! With larger pre-training corpus (e.g., 5 million <SQL, Table, Execution Result> pairs), the performance on downstream datasets would be better.
```
Comment (NielsRogge) [inline tapex.md:118]:
```suggestion
What about the efficiency? How efficient is such a pre-training method compared to the previous pre-training? The answer is given in the above figure: compared with previous table pre-training method TaBERT, TAPEX could yield 2% improvement only using 2% of the pre-training corpus, achieving a speedup of nearly **50** times! With a larger pre-training corpus (e.g., 5 million <SQL, Table, Execution Result> pairs), the performance on downstream datasets would be better.
```

File: tapex.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+
+### Conclusion
+
+In this blog, we introduce TAPEX, a table pre-training approach whose corpus is automatically synthesized via sampling SQL queries and their execution results. TAPEX addresses the data scarcity challenge in table pre-training by learning a neural SQL executor on a diverse, large-scale, and high-quality synthetic corpus. Experimental results on four downstream datasets demonstrate that TAPEX outperforms previous table pre-training approaches by a large margin, with a higher pre-training efficiency.
+
+### Take Away
+
+What can we learn from the success of TAPEX? I suggest that, especially if you want to perform efficient continual pre-training, you may try these options: (1) synthesize an accurate and small corpus, instead of mining a large but noisy corpus from Internet; (2) simulate domain-adaptive skills via programs, instead of general-purpose language modeling via natural language sentences.
```
Comment (NielsRogge) [inline tapex.md:127]:
```suggestion
What can we learn from the success of TAPEX? I suggest that, especially if you want to perform efficient continual pre-training, you may try these options: (1) synthesize an accurate and small corpus, instead of mining a large but noisy corpus from the Internet; (2) simulate domain-adaptive skills via programs, instead of general-purpose language modeling via natural language sentences. 
```

**Review by NielsRogge — APPROVED** [review]
Awesome blog! Thanks a lot for writing this.

File: tapex.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
+---
```
Comment (osanseviero) [inline tapex.md:1]: You need to also update https://github.com/huggingface/blog/blob/main/_blog.yml for the blog post to show

File: tapex.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+title: "Query Your Tables using Natural Language: An Introduction to TAPEX"
```
Comment (osanseviero) [inline tapex.md:2]: TAPEX is so cool :fire: maybe we could try a title that is more exciting :thinking:

File: tapex.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+<div class="blog-metadata">
+    <small>Published May 13, 2022.</small>
+    <a target="_blank" class="btn no-underline text-sm mb-5 font-sans" href="https://github.com/huggingface/blog/blob/main/tapex.md">
+        Update on GitHub
+    </a>
+</div>
+
+<div class="author-card">
+    <a href="https://github.com/SivilTaram">
```
Comment (osanseviero) [inline tapex.md:19]: we usually use the HF account, but if you prefer GH it's also ok!

File: tapex.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+        <img class="avatar avatar-user" src="https://aeiljuispo.cloudimg.io/v7/https://s3.amazonaws.com/moonup/production/uploads/1652435645515-612ee6a7b960e78c6d2319d4.jpeg?w=200&h=200&f=face" title="Gravatar">
+        <div class="bfc">
```
Comment (osanseviero) [inline tapex.md:21]: You can add a guest label as in https://github.com/huggingface/blog/blob/main/1b-sentence-embeddings.md?plain=1#L22

File: tapex.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+### Overview
+
+In "[TAPEX: Table Pre-training via Learning a Neural SQL Executor](https://openreview.net/forum?id=O50443AsCP)", we explore **using synthetic data as a proxy for real data during pre-training**, and demonstrate its powerfulness with *TAPEX (Table Pre-training via Execution)* as an example. In TAPEX, we show that table pre-training can be achieved by learning a neural SQL executor over a synthetic corpus.
+
+![snippet](assets/74_tapex/tapex-overview.png)
+> Note: [Table] is a placeholder for the user provided table in the input.
+
+As shown in the above figure, by systematically sampling *executable SQL queries and their execution outputs* over tables, TAPEX first synthesizes a synthetic and non-natural pre-training corpus. Then it continues to pre-train a language model (e.g., [BART](https://arxiv.org/abs/1910.13461)) to output the execution results of SQL queries, which mimics the process of a neural SQL executor.
```
Comment (osanseviero) [inline tapex.md:38]:
```suggestion
As shown in the figure above, by systematically sampling *executable SQL queries and their execution outputs* over tables, TAPEX first synthesizes a synthetic and non-natural pre-training corpus. Then, it continues to pre-train a language model (e.g., [BART](https://arxiv.org/abs/1910.13461)) to output the execution results of SQL queries, which mimics the process of a neural SQL executor.
```

File: tapex.md [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+
+print(tokenizer.batch_decode(outputs, skip_special_tokens=True))
+# ['2008']
+```
+
+### Fine-tuning
+
+During fine-tuning, we feed the concatenation of the natural language question and the flattened table to the model as input, the answer labeled by annotators serves as the supervision for the model as output. Want to fine-tune TAPEX by yourself? You can look at the fine-tuning script [here](https://github.com/huggingface/transformers/tree/main/examples/research_projects/tapex), which has been officially integrated into 🤗 Transformers `4.19.0`!
+
+And by now, all available fine-tuned TAPEX models have interactive widgets officially supported by Huggingface! You can try to answer some questions as below.
```
Comment (osanseviero) [inline tapex.md:77]: Here it would be cool to have a link to the hub with a filter for tapex models

File: tapex.md [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+
+
+### Conclusion
+
+In this blog, we introduce TAPEX, a table pre-training approach whose corpus is automatically synthesized via sampling SQL queries and their execution results. TAPEX addresses the data scarcity challenge in table pre-training by learning a neural SQL executor on a diverse, large-scale, and high-quality synthetic corpus. Experimental results on four downstream datasets demonstrate that TAPEX outperforms previous table pre-training approaches by a large margin, with a higher pre-training efficiency.
+
+### Take Away
+
+What can we learn from the success of TAPEX? I suggest that, especially if you want to perform efficient continual pre-training, you may try these options: (1) synthesize an accurate and small corpus, instead of mining a large but noisy corpus from the Internet; (2) simulate domain-adaptive skills via programs, instead of general-purpose language modeling via natural language sentences.
```
Comment (osanseviero) [inline tapex.md:127]: Given this is markdown I would suggest to have this as an ordered list

**Review by osanseviero — COMMENTED** [review]
Very cool blog post :fire: :fire: I left some suggestions but overall looks good!

File: tapex.md (hunk 8 above)
Comment (SivilTaram) [reply tapex.md:19]: Yeah, I prefer GH lol. Thanks for the suggestion!

File: tapex.md (hunk 9 above)
Comment (SivilTaram) [reply tapex.md:21]: Good catch! Done.

File: tapex.md (hunk 11 above)
Comment (SivilTaram) [reply tapex.md:77]: Great suggestion, thanks!

File: tapex.md (hunk 12 above)
Comment (SivilTaram) [reply tapex.md:127]: Got it, fixed!

File: tapex.md (hunk 7 above)
Comment (SivilTaram) [reply tapex.md]: Resolved!

File: _blog.yml [hunk 13]
```diff
@@ -811,3 +811,13 @@
   tags:
   - expert-acceleration-program
   - case-study
+
+- local: tapex
+  title: "Query Your Tables using Natural Language: An Introduction to TAPEX"
```
[the remark this answers is not in the document]
Comment (SivilTaram) [reply _blog.yml]: Good catch! Fix that now, also with the author name.

**Review by osanseviero — APPROVED** [review]
Looks great :fire:
